mxtools/flyer.py

```diff
@@ -167,98 +167,147 @@ def unstage(self):
-        vector,
-        zebra,
-        eiger,
+        PW = (exposurePeriodPerImage - detector_dead_time) * 1000.0
+        PS = (exposurePeriodPerImage) * 1000.0
+        GW = scanWidth - (1.0 - (PW / PS)) * (imgWidth / 2.0)
+        self.setup_zebra_vector_scan(
+            angle_start=angle_start,
+            gate_width=GW,
+            scan_width=scanWidth,
+            pulse_width=PW,
+            pulse_step=PS,
+            exposure_period_per_image=exposurePeriodPerImage,
+            num_images=numImages,
+            is_still=imgWidth == 0,
+        )
+
+    def detector_arm(self, **kwargs):
+        start = kwargs["start"]
+        width = kwargs["width"]
+        num_images = kwargs["num_images"]
+        exposure_per_image = kwargs["exposure_per_image"]
+        file_prefix = kwargs["file_prefix"]
+        data_directory_name = kwargs["data_directory_name"]
+        file_number_start = kwargs["file_number_start"]
+        x_beam = kwargs["x_beam"]
+        y_beam = kwargs["y_beam"]
+        wavelength = kwargs["wavelength"]
+        det_distance_m = kwargs["det_distance_m"]
+
+        self.detector.cam.save_files.put(1)
+        self.detector.cam.file_owner.put(getpass.getuser())
+        self.detector.cam.file_owner_grp.put(grp.getgrgid(os.getgid())[0])
+        self.detector.cam.file_perms.put(420)
+        file_prefix_minus_directory = str(file_prefix)
+        file_prefix_minus_directory = file_prefix_minus_directory.split("/")[-1]
+
+        self.detector.cam.acquire_time.put(exposure_per_image)
+        self.detector.cam.acquire_period.put(exposure_per_image)
+        self.detector.cam.num_images.put(num_images)
+        self.detector.cam.file_path.put(data_directory_name)
+        self.detector.cam.fw_name_pattern.put(f"{file_prefix_minus_directory}_$id")
+
+        # TODO: change it back to detector.cam.sequence_id once the ophyd PR
+        # https://github.com/bluesky/ophyd/pull/1001 is merged/released.
+        self.detector.file.sequence_id.put(file_number_start)
+
+        # originally from detector_set_fileheader
+        self.detector.cam.beam_center_x.put(x_beam)
+        self.detector.cam.beam_center_y.put(y_beam)
+        self.detector.cam.omega_incr.put(width)
+        self.detector.cam.omega_start.put(start)
+        self.detector.cam.wavelength.put(wavelength)
+        self.detector.cam.det_distance.put(det_distance_m)
+
+        start_arm = ttime.time()
+        self.detector.cam.acquire.put(1)
+        logger.info(f"arm time = {ttime.time() - start_arm}")
+
+    def setup_vector_program(self, num_images, angle_start, angle_end, exposure_period_per_image):
+        self.vector.num_frames.put(num_images)
+        self.vector.start.omega.put(angle_start)
+        self.vector.end.omega.put(angle_end)
+        self.vector.frame_exptime.put(exposure_period_per_image * 1000.0)
+        self.vector.hold.put(0)
+
+    def zebra_daq_prep(self):
+        self.zebra.reset.put(1)
+        ttime.sleep(2.0)
```

Comment:
We don't need to do this asynchronously (ie. yield from)?

Reply:
nope, Maksim recommended that we do it this way. this and the following question are answered by looking at the pseudocode show in issue 137: https://github.com/NSLS-II/lsdc/issues/137 
`flyer.update_parameters(**kwargs)`
`RE(bps.fly([flyer])`

These calls would need to be done in either test code or within LSDC itself.